### PR TITLE
CBMC proof for hash_table_swap()

### PR DESCRIPTION
**Pull request by danielsn**
*Description of changes:*
CBMC proof for ```hash_table_swap()```.  

Due to the fact that either input to swap can be invalid, it's hard to come up with any good post-conditions for this function.  It might be possible to track some using ghost state, but that would put unnecessary overhead on the function itself.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/hash_table.c [hunk 1]
```diff
@@ -231,6 +231,7 @@ void aws_hash_table_clean_up(struct aws_hash_table *map) {
 }
 
 void aws_hash_table_swap(struct aws_hash_table *AWS_RESTRICT a, struct aws_hash_table *AWS_RESTRICT b) {
+    AWS_PRECONDITION(a != b);
```
Comment (feliperodri) [inline source/hash_table.c]:
```
AWS_PRECONDITION(aws_hash_table_is_valid(a));
AWS_PRECONDITION(aws_hash_table_is_valid(b));
```
?

**Comment by danielsn** [comment]
I had the same thought at first. But the spec for swap requires neither
input to be valid as per the .h file.

File: .cbmc-batch/jobs/aws_hash_table_swap/aws_hash_table_swap_harness.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+void aws_hash_table_swap_harness() {
+    struct aws_hash_table a;
+    struct aws_hash_table b;
+    bool inita;
+    bool initb;
+    struct store_byte_from_buffer stored_byte_a;
+    struct store_byte_from_buffer stored_byte_b;
+
+    // There are no loops in the code under test, so use the biggest possible value
```
Comment (JonathanHenson) [inline .cbmc-batch/jobs/aws_hash_table_swap/aws_hash_table_swap_harness.c]: c++ comments...

Comment (danielsn) [reply .cbmc-batch/jobs/aws_hash_table_swap/aws_hash_table_swap_harness.c]: Fixed.

**Pull request merged**